src/vault.py

```diff
@@ -22,8 +22,14 @@ class VaultError(Exception):
 class Vault:
     """Class to interact with Vault through its API."""
 
-    def __init__(self, url: str, ca_cert_path: str):
-        self._client = hvac.Client(url=url, verify=ca_cert_path)
+    def __init__(self, url: str, auth_details: Dict[str, str]):
+        if ca_cert := auth_details.get("ca-cert-path"):
+            if (cert := auth_details.get("cert-path")) and (pk := auth_details.get("key-path")):
+                self._client = hvac.Client(url=url, cert=(cert, pk), verify=ca_cert)
+            else:
+                self._client = hvac.Client(url=url, verify=ca_cert)
```

Comment:
Why this complexity? Do we really need multiple ways of instantiating the vault client? Even if we do I'd recommend having multiple attributes in the init instead of a dict of str to str that could contain anything.

Reply:
I wanted to make it extensible but I think right now it was just more trouble than it was worth. I've reverted all of the changes related to this.